x/authz/client/cli/tx.go

```diff
@@ -0,0 +1,185 @@
+				return err
+			}
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	cmd.Flags().Int64(FlagExpiration, int64(3600*24*365), "The second unit of time duration which the authorization is active for the user; Default is a year")
+	return cmd
+}
+
+func NewCmdRevokeAuthorization() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "revoke [grantee_address] [msg_type] --from=[granter_address]",
+		Short: "revoke authorization",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`revoke authorization from a granter to a grantee:
+Example:
+ $ %s tx %s revoke cosmos1skj.. %s --from=cosmos1skj..
+			`, version.AppName, types.ModuleName, types.SendAuthorization{}.MethodName()),
+		),
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			grantee, err := sdk.AccAddressFromBech32(args[0])
+			if err != nil {
+				return err
+			}
+
+			granter := clientCtx.GetFromAddress()
+
+			msgAuthorized := args[1]
+
+			msg := types.NewMsgRevokeAuthorization(granter, grantee, msgAuthorized)
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+func NewCmdExecAuthorization() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "exec [msg_tx_json_file] --from [grantee]",
+		Short: "execute tx on behalf of granter account",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`execute tx on behalf of granter account:
+Example:
+ $ %s tx %s exec tx.json --from grantee
+ $ %s tx bank send <granter> <recipient> --from <granter> --chain-id <chain-id> --generate-only > tx.json && %s tx %s exec tx.json --from grantee
+			`, version.AppName, types.ModuleName, version.AppName, version.AppName, types.ModuleName),
+		),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+			grantee := clientCtx.GetFromAddress()
+
+			if offline, _ := cmd.Flags().GetBool(flags.FlagOffline); offline {
+				return errors.New("cannot broadcast tx during offline mode")
+			}
+
+			stdTx, err := authclient.ReadTxFromFile(clientCtx, args[0])
+			if err != nil {
+				return err
+			}
+			msgs := stdTx.GetMsgs()
+			serviceMsgs := make([]sdk.ServiceMsg, len(msgs))
+			for i, msg := range msgs {
+				serviceMsgs[i] = msg.(sdk.ServiceMsg)
+			}
+
+			msg := types.NewMsgExecAuthorized(grantee, serviceMsgs)
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
```

Comment:
I think this might work right now, but is not ideal. `types.NewMsgExecAuthorized` returns a `MsgExecAuthorized` (the request type, not the service Msg).

I would propose to wait for https://github.com/cosmos/cosmos-sdk/issues/7541 to avoid refactoring, or do something like the temporary [MsgSend service Msg](https://github.com/cosmos/cosmos-sdk/blob/2f1d8771df028a1c7844dd8218a4db6b0f56ba7d/x/bank/client/testutil/cli_helpers.go#L95-L103)